backend/infrahub/git/repository.py

```diff
@@ -170,7 +171,12 @@ async def merge(self, source_branch: str, dest_branch: str, push_remote: bool =
         commit = self.get_commit_value(branch_name=source_branch, remote=False)
 
         try:
-            repo.git.merge(commit)
+            if config.SETTINGS.git.allow_explicit_merge_commit:
+                if not config.SETTINGS.git.user_name:
+                    raise ValueError("Cannot allow explicit merge commit if git user_name setting is not set.")
```

Comment:
What would be the user behavior when we raise `ValueError` here?
How would the user become aware that there was an issue and what action would we expect the user to take?

Reply:
I'd suggest that we ignore this setting and if not configured we'll just use the default username that's set in the Docker image.